File: Homework6/fileManager/PhoneBookManager.java
package Homework6.fileManager;

import Homework6.fileManager.reader.PhoneBookReader;
import Homework6.fileManager.writer.PhoneBookWriter;
import Homework6.model.Contact;

import java.util.List;

public class PhoneBookManager {
    private PhoneBookWriter writer;
    private PhoneBookReader reader;

    public PhoneBookManager(PhoneBookWriter writer, PhoneBookReader reader) {
        this.writer = writer;
        this.reader = reader;
    }

    public void saveContact(Contact contacts) {
        writer.save(contacts);
    }

    public List<Contact> readContacts() {
        return reader.readPhoneBook();
    }
}

File: Homework7/fileManager/PhoneBookManager.kt
package Homework7.fileManager

import Homework7.fileManager.reader.PhoneBookReader
import Homework7.fileManager.writer.PhoneBookWriter
import Homework7.model.Contact

class PhoneBookManager(private val writer: PhoneBookWriter, private val reader: PhoneBookReader) {
    fun saveContact(contacts: Contact) {
        writer.save(contacts)
    }

    fun readContacts(): MutableList<Contact> {
        return reader.readPhoneBook()
    }
}